cli/compose/loader/loader.go

```diff
@@ -62,14 +85,18 @@ func Load(configDetails types.ConfigDetails) (*types.Config, error) {
 			return nil, err
 		}
 
-		var err error
-		configDict, err = interpolateConfig(configDict, configDetails.LookupEnv)
-		if err != nil {
-			return nil, err
+		if !opts.SkipInterpolation {
+			var err error
+			configDict, err = interpolateConfig(configDict, *opts.Interpolate)
+			if err != nil {
+				return nil, err
+			}
 		}
 
-		if err := schema.Validate(configDict, configDetails.Version); err != nil {
-			return nil, err
+		if !opts.SkipValidation {
```

Comment:
I guess the reason to allow skipping "validation" is that if "interpolation" is skipped, there's values that are invalid (because, e.g. `$port` is not a valid port-number)?

Is there a use-case to disable `validation` or `interpolation` separately?

Also, trying to think of situations where skipping validation could skip important checks (although, I guess in the end that's the daemon/API's responsibility)

Reply:
If we don't interpolate, for sure we don't want to validate (otherwise the composefile could be invalid). The other way could be *happens*, i.e. doing some interpolation but not validating (for whatever reason, like if it's later on used with something else).

> Also, trying to think of situations where skipping validation could skip important checks (although, I guess in the end that's the daemon/API's responsibility)

Yes. And also, this is *only* in the library part of `cli/compose` — i.e. with this PR it is not possible to not validate or interpolate (and we shouldn't allow that with the `docker` cli) :wink: